src/script/cpatch.py

```diff
@@ -0,0 +1,727 @@
+                    self._copy_binary(path, lib_dst / path.name)
+        except FileNotFoundError as err:
+            log.warning("skipping lib %s: %s", name, err)
+        return f"ADD {name} {destination}"
+
+    def _py_site_packages(self):
+        """Return the correct python site packages dir for the image."""
+        if self._cached_py_site_packages is not None:
+            return self._cached_py_site_packages
+        # use the container image to probe for the correct python site-packages dir
+        valid_site_packages = [
+            "/usr/lib/python3.8/site-packages",
+            "/usr/lib/python3.6/site-packages",
+        ]
+        cmd = [
+            self._ctx.engine,
+            "run",
+            "--rm",
+            self._ctx.base_image,
+            "ls",
+            "-d",
+        ]
+        cmd += valid_site_packages
+        if self._ctx.root_build:
+            cmd.insert(0, "sudo")
+        result = _run(cmd, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
+        log.debug(f"container output: {result.stdout!r}")
+        for line in result.stdout.decode("utf8").splitlines():
+            pdir = line.strip()
+            if line.strip() in valid_site_packages:
+                log.debug(f"found site packages dir: {pdir}")
+                self._cached_py_site_packages = pdir
+                return self._cached_py_site_packages
+        raise ValueError(
+            "no valid python site-packages dir found in container"
+        )
+
+    def _py_mgr_job(self, component):
+        name = "mgr_plugins.tar"
+        if self._include_dashboard:
+            log.debug("Including dashboard with mgr")
+            exclude_dirs = ("tests",)
+        else:
+            log.debug("Excluding dashboard from mgr")
+            exclude_dirs = ("tests", "node_modules")
+        exclude_file_suffixes = (".pyc", ".pyo", ".tmp", "~")
+        with tarfile.open(self._workdir / name, mode="w") as tar:
+            with ChangeDir(self._ctx.source_dir / "src/pybind/mgr"):
+                self._build_tar(
+                    tar,
+                    exclude_dirs=exclude_dirs,
+                    exclude_file_suffixes=exclude_file_suffixes,
+                )
+        return [f"ADD {name} /usr/share/ceph/mgr"]
+
+    def _py_common_job(self, component):
+        name = "python_common.tar"
+        exclude_dirs = ("tests", "node_modules")
+        exclude_file_suffixes = (".pyc", ".pyo", ".tmp", "~")
+        with tarfile.open(self._workdir / name, mode="w") as tar:
+            with ChangeDir(self._ctx.source_dir / "src/python-common"):
+                self._build_tar(
+                    tar,
+                    exclude_dirs=exclude_dirs,
+                    exclude_file_suffixes=exclude_file_suffixes,
+                )
+        return [f"ADD {name} {self._py_site_packages()}"]
+
+    def _cephadm_job(self, component):
+        # TODO: support "compiled" cephadm
```

Comment:
Is this still "TODO"? It seems like the `else` case below is for the compiled cephadm case.

Reply:
No, I am using it all the time with "compiled" cephadm.

Reply:
Fixed